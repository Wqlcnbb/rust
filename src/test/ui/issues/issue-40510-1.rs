#![allow(unused)]

fn f() {
    let mut x: Box<()> = Box::new(());

    || {
        &mut x //~ ERROR cannot infer
    };
}


fn main() {}
