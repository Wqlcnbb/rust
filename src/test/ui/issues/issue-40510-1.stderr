error[E0495]: cannot infer an appropriate lifetime for borrow expression due to conflicting requirements
  --> $DIR/issue-40510-1.rs:7:9
   |
LL |         &mut x
   |         ^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime '_ as defined on the body at 6:5...
  --> $DIR/issue-40510-1.rs:6:5
   |
LL |     || {
   |     ^^
note: ...so that closure can access `x`
  --> $DIR/issue-40510-1.rs:7:9
   |
LL |         &mut x
   |         ^^^^^^
note: but, the lifetime must be valid for the expression at 6:5...
  --> $DIR/issue-40510-1.rs:6:5
   |
LL | /     || {
LL | |         &mut x
LL | |     };
   | |_____^
note: ...so type `[closure@$DIR/issue-40510-1.rs:6:5: 8:6 x:&mut std::boxed::Box<()>]` of expression is valid during the expression
  --> $DIR/issue-40510-1.rs:6:5
   |
LL | /     || {
LL | |         &mut x
LL | |     };
   | |_____^

error: aborting due to previous error

